File: presentation/api/v2/schemas/subscriptions.py
"""Subscription/account schemas."""

from typing import Optional, List
from pydantic import BaseModel


class SubscriptionInfo(BaseModel):
    subscription_id: str
    subscription_name: str
    tenant_id: Optional[str] = None
    state: str = "Enabled"
    is_default: bool = False


class SubscriptionListResponse(BaseModel):
    subscriptions: List[SubscriptionInfo]
    total: int

File: frontend/src/api/subscriptions.ts
import { api } from './client'

interface Subscription {
  subscription_id: string
  subscription_name: string
  tenant_id?: string
  state: string
  is_default: boolean
}

export interface DropdownSubscription {
  id: string
  name: string
  is_default: boolean
}

export async function getSubscriptions(): Promise<{ subscriptions: Subscription[]; total: number }> {
  return api.get('/api/subscriptions')
}

export async function getSubscriptionDropdown(provider: string = 'azure'): Promise<{ subscriptions: DropdownSubscription[] }> {
  return api.get(`/api/subscriptions/dropdown?provider=${encodeURIComponent(provider)}`)
}

export async function validateSubscription(subscriptionId: string) {
  return api.post(`/api/subscriptions/${encodeURIComponent(subscriptionId)}/validate`)
}
